src/WebSockets.jl

```diff
@@ -384,6 +384,7 @@ function open(f::Function, url; suppress_close_error::Bool=false, verbose=false,
                     close(ws, CloseFrameBody(1008, "Unexpected client websocket error"))
                 end
             end
+            rethrow(e)
```

Comment:
is `rethrow()` fine here? 
```suggestion
            rethrow()
```
I think `rethrow(e)` is not usually the right choice, per the docs:

```julia
help?> rethrow
search: rethrow

  rethrow()

  Rethrow the current exception from within a catch block. The rethrown exception will continue propagation as if it had not been caught.

  │ Note
  │
  │  The alternative form rethrow(e) allows you to associate an alternative exception object e with the current backtrace. However this
  │  misrepresents the program state at the time of the error so you're encouraged to instead throw a new exception using throw(e). In Julia
  │  1.1 and above, using throw(e) will preserve the root cause exception on the stack, as described in current_exceptions.

```

Reply:
I think we should actually do the rethrowing from inside the `!isok` block above, since we're currently "throwing" a "closed normally" exception when we get a close notification from the peer socket.